### DISCUSSION TITLE
Client js Monitoring in Skywalking

### DISCUSSION THREAD

**Discussion by Maurya243** [apache/skywalking#12016] [Q&A]
Hi Team,

Could you please tell us the procedure on how to do client js monitoring in skywalking ?

**Comment by Maurya243** [comment]
We tried with a single page html code but not showing data in skywalking which is of latetst version. Please find the attached screenshots of skywalking and single page application.
![Single page](https://github.com/apache/skywalking/assets/66820660/ca4febbe-39b7-4a05-a826-ec3172dff2de)
![Skywalking](https://github.com/apache/skywalking/assets/66820660/23e85494-9557-454d-abe6-cee74fd3cff2)
Below is the html code where we have provided the script into it : **_(test.html)_**
"
_<html>
<head>
<title>instna test</title>
<script type="module">
import ClientMonitor from 'C:/ClientjsMonitoring/node_modules/skywalking-client-js/';
ClientMonitor.register({
  collector: 'http://10.83.100.53:12800',
  service: 'Hyphen_View',
  pagePath: 'test.html',
  serviceVersion: 'v1.0.0',
});
</script>
</head>
<body>
<form>
<h1>Test Client-JS </h1>
<button id="bt1" onclick="myfun()"> click to see</button>
</form>
<script>

function myfun(){
alert("hii");
console.lag("ah");
}

</script>
</body>
</html>_
"
![skywalking-client-js](https://github.com/apache/skywalking/assets/66820660/c02a8c3f-90ae-4b07-8d9b-7e4117598d44)
![application folder](https://github.com/apache/skywalking/assets/66820660/a5083992-f920-4d72-a3f1-c94ef5f12cd8)

Also we tried with full path and half path under _import ClientMonitor_ but still throwing error and data is not reflecting in skywalking.

**Reply by Fine0830** [reply]
It seems to be a cross origin problem.

**Reply by Maurya243** [reply]
Thank you for your response.
Is it possible to resolve the issue or is there any alternative solution to resolve the same?

**Reply by tanshanshan** [reply]
Cross-domain problem,
you can install nginx and then use nginx proxy to access the backend of Skywalking.
such as

```
 location /browser {
            proxy_pass  http://ip:12800;
        }
```

```
        location /v3 {
            proxy_pass  http://ip:12800;
        }
```

**Comment by Maurya243** [comment]
Attaching the snapshot of text.html code 
![testhtml](https://github.com/apache/skywalking/assets/66820660/2527f54a-ff38-448e-89be-7a08885314a0)

**Reply by Fine0830** [reply]
Can you import `clientjs` like `ClientMonitor from 'skywalking-client-js';`?

**Reply by Maurya243** [reply]
Thank you for your response,
I hav tried with suggested method but still getting the same error as per the below snapshots.
![html2](https://github.com/apache/skywalking/assets/66820660/843fe53f-0c1e-4a55-b4ec-5753ee1e4394)
![test2](https://github.com/apache/skywalking/assets/66820660/f241eb63-5430-4330-a5f3-5c98741ae26f)
![test1](https://github.com/apache/skywalking/assets/66820660/23091d11-eb49-43df-883f-abbfadc1bc4d)
![html1](https://github.com/apache/skywalking/assets/66820660/2f70d21a-7958-4676-a0a7-280ba965795d)

**Reply by wu-sheng** [reply]
Why do you read a local HTML file? client-js is a lib monitoring pages hosted on web server.

**Reply by Maurya243** [reply]
We just tried to monitor the single html page using client js library. If that wont work can you please help us out to do client js monitoring in hyphendiag?

**Reply by wu-sheng** [reply]
This is not a single page application, this error logs show it is a local file.

**Reply by wu-sheng** [reply]
Why should we know hyphendiag? I am confused, we are not hyphendiag developers or vendors.

**Reply by Maurya243** [reply]
Sorry for the mistake in my late response,

Can you please provide the steps on how to configure the client js monitoring in skywalking?

**Reply by wu-sheng** [reply]
We provided all we have in the readme doc. For most users, that is enough.
